engine/gamesys/src/gamesys/test/test_gamesys.cpp

```diff
@@ -1091,6 +1091,53 @@ TEST_P(CollectionFactoryTest, Test)
 
 /* Draw Count */
 
+void AssertVertexEqual(const dmGameSystem::BoxVertex& lhs, const dmGameSystem::BoxVertex& rhs)
+{
+    static const float test_epsilon = 0.000001f;
+    EXPECT_NEAR(lhs.m_Position[0], rhs.m_Position[0], test_epsilon);
+    EXPECT_NEAR(lhs.m_Position[1], rhs.m_Position[1], test_epsilon);
+    EXPECT_NEAR(lhs.m_UV[0], rhs.m_UV[0], test_epsilon);
+    EXPECT_NEAR(lhs.m_UV[1], rhs.m_UV[1], test_epsilon);
+}
+
+TEST_P(BoxRenderTest, BoxRender)
+{
+    const BoxRenderParams& p = GetParam();
+    const char* go_path = p.m_GOPath;
+
+    ASSERT_TRUE(dmGameObject::Init(m_Collection));
+
+    // Spawn the game object with the script we want to call
+    dmGameObject::HInstance go = Spawn(m_Factory, m_Collection, go_path, dmHashString64("/go"), 0, 0, Point3(0, 0, 0), Quat(0, 0, 0, 1), Vector3(1, 1, 1));
+    ASSERT_NE((void*)0, go);
+
+    ASSERT_TRUE(dmGameObject::Update(m_Collection, &m_UpdateContext));
+
+    // Make the render list that will be used later.
+    dmRender::RenderListBegin(m_RenderContext);
+
+    dmGameSystem::GuiWorld* world = (dmGameSystem::GuiWorld*)m_GuiContext.m_Worlds[0];
+    dmGui::SetSceneAdjustReference(world->m_Components[0]->m_Scene, dmGui::ADJUST_REFERENCE_DISABLED);
+
+    dmGameObject::Render(m_Collection);
+
+    dmRender::RenderListEnd(m_RenderContext);
+    dmRender::DrawRenderList(m_RenderContext, 0x0, 0x0);
+
+    ASSERT_EQ(world->m_ClientVertexBuffer.Size(), (u_int32_t)p.m_ExpectedVerticesCount);
```

Comment:
What's this type "u_int32_t" ?
Is it just a typo?

Where is "m_ExpectedVerticesCount" set?

Reply:
Yes, it is a typo, but a funny one, cause apparently OSX has this type defined also :) Fixed in: https://github.com/defold/defold/pull/4921/commits/ee8b54705e82bd3f14a3c75619f23bdeaf0f2fa0